---
title: "Bifurcated Relationship between Amyloid and Tau Clearance"
author: "Julie Wisch"
date: "6/16/2021"
output:
  html_document:
    toc: true
    toc_depth: 4
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(warning = FALSE)
knitr::opts_chunk$set(message = FALSE)
library(readxl)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(grid)
library(lme4)
library(table1)
library(plotly)
library(data.table)
library(RobMixReg)
library(lcmm)
library(tidyr)
library(pROC)
library(ggROC)
library(caret)
library(supclust)
library(imputeMissings)
library(glmnet)
library(heatmaply)
library(ape)
library(survival)

library('Cairo')
CairoWin() #dealing with aliasing

source(".././R/functions.R")




pet <- read_xlsx(".././Data/mod_pet.xlsx")
mmse <- read_xlsx(".././Data/mod_b4_cdr.xlsx")
demogs <- read_xlsx(".././Data/mod_demographics.xlsx")
csf <- read_xlsx(".././Data/mod_lumipulse.xlsx")
np <- read.csv(".././Data/mod_psychometrics.csv")
mri <- read_xlsx(".././Data/mod_mri.xlsx")
tau <- read_xlsx(".././Data/mod_tau.xlsx")
wmh <- read_xlsx(".././Data/mod_wmh.xlsx")
nfl <- read.csv(".././Data/csf_nfl_20210301.csv")

csf_long <- setDT(csf)[, .N, by = ID]
csf_long <- csf_long[csf_long$N >= 4,]
#csf_long <- csf[csf$ID %in% csf_long$ID,] #making it so we keep everyone with 4 + LP's

csf <- csf[, c("ID", "CSF_LP_DATE", "LUMIPULSE_CSF_AB42", "LUMIPULSE_CSF_AB40",
               "LUMIPULSE_CSF_pTau", "LUMIPULSE_CSF_tTau")]
csf$LUMIPULSE_CSF_AB42_AB40 <- csf$LUMIPULSE_CSF_AB42 / csf$LUMIPULSE_CSF_AB40
csf$CSF_LP_DATE <- as.Date(csf$CSF_LP_DATE, format = "%Y-%m-%d")
csf <- csf[csf$ID %in% csf_long$ID,]
```

When I started looking at longitudinal amyloid accumulation, I noticed a weird bifurcation in the relationship between pTau and AB42. When you look at the plot below (plotting all individuals in our ADRC dataset with 4 or more lumbar punctures), you can see around AB42 = 750 that there is a group of participants who have a very steep increase in pTau levels, and another group of participants who show a steady decrease in pTau levels.

```{r}
ggplot(csf, aes(x = LUMIPULSE_CSF_AB42, y = LUMIPULSE_CSF_pTau, group = ID))+ geom_point(alpha = 0.8) +geom_line(alpha = 0.8) +
  #geom_smooth(method = "gam", aes(group = 1), colour = "black") + 
  theme_bw() +
  scale_x_reverse() + geom_vline(xintercept = 750, colour = "red")
```

In order to separate participants based on these observed trajectories, I applied Growth Mixture Modeling (R Package lcmm). Growth Mixture Modeling takes into account repeated measures and allows the model to vary by both slope and intercept for each individual. Because all of our participants had completed at least 4 lumbar punctures, it felt reasonable to allow for varying slopes. While I judged by looking at the above figure that there were two distinct groups, I ran the growth mixture models on 1, 2, 3 and 4 latent classes to be sure that my personal judgment aligned with the optimized classification. The model formulation is shown below. 

```{r hlme_model_run_vary_slope, cache = TRUE}
set.seed(2002)
gmm1_2 <- hlme(LUMIPULSE_CSF_pTau ~ LUMIPULSE_CSF_AB42, subject = "ID", random=~1+LUMIPULSE_CSF_AB42, ng = 1, data =
               csf, verbose = FALSE)
gmm2_2 <- gridsearch(rep = 100, maxiter = 10, minit = gmm1_2,
                   hlme(LUMIPULSE_CSF_pTau ~ LUMIPULSE_CSF_AB42, subject = "ID", random=~1+LUMIPULSE_CSF_AB42,
                        ng = 2, data = csf, mixture = ~ LUMIPULSE_CSF_AB42,
                        nwg=T, verbose = FALSE))
gmm3_2 <- gridsearch(rep = 100, maxiter = 10, minit = gmm1_2,
                   hlme(LUMIPULSE_CSF_pTau ~ LUMIPULSE_CSF_AB42, subject = "ID", random=~1+LUMIPULSE_CSF_AB42,
                        ng = 3, data = csf, mixture = ~ LUMIPULSE_CSF_AB42,
                        nwg=T, verbose = FALSE))
gmm4_2 <- gridsearch(rep = 100, maxiter = 10, minit = gmm1_2,
                   hlme(LUMIPULSE_CSF_pTau ~ LUMIPULSE_CSF_AB42, subject = "ID", random=~1+LUMIPULSE_CSF_AB42,
                        ng = 4, data = csf, mixture = ~ LUMIPULSE_CSF_AB42,
                        nwg=T, verbose = FALSE))
```

Instead of 2 classes, the growth mixture model identified 3 distinct latent clusters of individuals on the amyloid-tau trajectory. This was determined by finding the model that minimized BIC.

```{r}
summarytable(gmm1_2, gmm2_2, gmm3_2, gmm4_2)
```

When we plot these individuals, we see that cluster 1 is somewhat transitional, cluster 2 is individuals with skyrocketing tau levels and low amyloid clearance, and cluster 3 is individuals with low amyloid clearance and low tau levels. Insel et al, 2017 (https://doi.org/10.3389/fnins.2017.00281) talk about "emerging AB pathology" as the time when individuals have early stage abnormal amyloid metabolism prior to meeting conventional preclinical AD criteria. We will use that terminology here.

```{r}
selected_cluster <- data.frame("ID" = gmm3_2$pprob$ID, "upvoted_cluster" = gmm3_2$pprob$class)
results <- merge(csf, selected_cluster, by = "ID")
results$upvoted_cluster <- as.factor(results$upvoted_cluster)
levels(results$upvoted_cluster) <- c("Intermediate AD Biomarkers", "AD Biomarker Positive", "AD Biomarker Negative")
g1 <- ggplot(results, aes(x = LUMIPULSE_CSF_AB42, y = LUMIPULSE_CSF_pTau, group = ID, colour = upvoted_cluster))+ 
  geom_point(alpha = 0.8) +geom_line(alpha = 0.8) +
  geom_smooth(method = "gam", aes(group = 1), colour = "black") + theme_bw() +
  labs(colour = "Assigned Cluster", x = paste("CSF A", expression(beta), "42", sep = ""), 
       y = "CSF pTau")+
  scale_x_reverse() + scale_colour_viridis(discrete = TRUE)


```

Above we show a single GAM fitted across all longitudinal visits. Below we show individual GAMs plotted for each cluster of participants.

```{r}
cluster_results <- unique(results[, c("ID", "upvoted_cluster")])

clustered_plot <- ggplot(results, aes(x = LUMIPULSE_CSF_AB42, y = LUMIPULSE_CSF_pTau, group = ID, colour = upvoted_cluster))+ 
  geom_point(alpha = 0.8) +geom_line(alpha = 0.8) +
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) + theme_bw() +
  labs(colour = "Assigned Cluster", x = paste("CSF A", expression(beta), "42", sep = ""), 
       y = "CSF pTau")+
  scale_x_reverse() + scale_colour_viridis(discrete = TRUE)

results$AB42_AB40 <- results$LUMIPULSE_CSF_AB42 / results$LUMIPULSE_CSF_AB40
clustered_plot_ratio <- ggplot(results, aes(x = AB42_AB40, y = LUMIPULSE_CSF_pTau, group = ID, colour = upvoted_cluster))+ 
  geom_point(alpha = 0.8) +geom_line(alpha = 0.8) +
  geom_smooth(method = "gam", aes(group = 1), colour = "black") + theme_bw() +
  labs(colour = "Assigned Cluster", x = paste("CSF A", expression(beta), "42 / CSF A", expression(beta), "40", sep = ""), 
       y = "CSF pTau")+
  scale_x_reverse() + scale_colour_viridis(discrete = TRUE)

clustered_plot

g <- lemon::grid_arrange_shared_legend(clustered_plot, clustered_plot_ratio, nrow = 2, ncol = 1)

ggsave(g, filename = 'Fig1.png', dpi = 300, type = 'cairo',
       width = 8, height = 6, units = 'in')
```
Here we extract the actual GAMMs for the three groups. I'm doing this so I can get the mean value for the Emerging Pathology guys once they are amyloid positive (CSF AB42 < 1098).

```{r}

gam_from_plot <- ggplot_build(clustered_plot)$data[[3]]

print("Model forecasts for the AD cohort")
paste0("Mean = ", mean(gam_from_plot[gam_from_plot$group == 2 & gam_from_plot$x > -1098, "y"]))
paste0("SE = ", mean(gam_from_plot[gam_from_plot$group == 2 & gam_from_plot$x > -1098, "se"]))
paste0("SD = ", sd(gam_from_plot[gam_from_plot$group == 2 & gam_from_plot$x > -1098, "y"]))

print("Model forecasts for the emerging cohort")
paste0("Mean = ", mean(gam_from_plot[gam_from_plot$group == 1 & gam_from_plot$x > -1098, "y"]))
paste0("SE = ", mean(gam_from_plot[gam_from_plot$group == 1 & gam_from_plot$x > -1098, "se"]))
paste0("SD = ", sd(gam_from_plot[gam_from_plot$group == 1 & gam_from_plot$x > -1098, "y"]))


print("Model forecasts for the Pathology Free cohort")
paste0("Mean = ", mean(gam_from_plot[gam_from_plot$group == 3 & gam_from_plot$x > -1098, "y"]))
paste0("SE = ", mean(gam_from_plot[gam_from_plot$group == 3 & gam_from_plot$x > -1098, "se"]))
paste0("SD = ", sd(gam_from_plot[gam_from_plot$group == 3 & gam_from_plot$x > -1098, "y"]))


```


Now we will compare the trajectories of these individuals across the A-T-N-Cognition spectrum and see how they differ.

```{r}
demogs$BIRTH <- as.Date(demogs$BIRTH, format = "%Y-%m-%d")
demogs <- demogs[, c("ID", "BIRTH", "GENDER", "EDUC", "apoe", "race2")]
demogs$apoe4 <- ifelse(demogs$apoe == 34 | demogs$apoe == 44, 1, 0)
demogs <- demogs[demogs$ID %in% unique(csf_long$ID),]
demogs <- merge(demogs, cluster_results, by = "ID")



csf <- merge(csf, demogs, by = "ID")
csf$Age_at_LP <- round(as.numeric(csf$CSF_LP_DATE - csf$BIRTH) / 365.25, 1)
nfl$LP_DATE <- as.Date(nfl$LP_DATE, format = "%m/%d/%Y")
csf <- merge(csf, nfl, by.x = c("ID", "CSF_LP_DATE"), by.y = c("ID", "LP_DATE"), all.x = TRUE, all.y = FALSE)

pet<- pet[!(pet$PUP_QC_Status == "Failed" | pet$PUP_QC_Status == "Quarantined"), ]
pet <- pet[pet$Tracer == "PIB",]
pet <- pet[, c("ID", "PET_Date", "pib_fsuvr_rsf_tot_cortmean")]
pet$PET_Date <- as.Date(pet$PET_Date, format = "%Y-%m-%d")
pet <- pet[!is.na(pet$pib_fsuvr_rsf_tot_cortmean),]
pet <- merge(pet, demogs, by = "ID", all = FALSE)
pet$Age_at_Scan <- as.numeric(pet$PET_Date - pet$BIRTH) / 365.25


mmse <- mmse[, c("ID", "testdate", "cdr", "MMSE", "sumbox", "memory", "orient", "judgment", "commun", "homehobb", "perscare")]
mmse$testdate <- as.Date(mmse$testdate, format = "%Y-%m-%d")
mmse <- mmse[!is.na(mmse$MMSE),]
mmse <- merge(mmse, demogs, by = "ID", all = FALSE)
mmse$Age_at_Visit <- as.numeric(mmse$testdate - mmse$BIRTH) / 365.25


np <- np[, c("ID", "psy_date", "srttotal", "lmdelay", "digsym", "MEMUNITS")]
np$psy_date <- as.Date(np$psy_date, format = "%d-%b-%y")
np <- merge(np, demogs, by = "ID")
np$Age_at_Visit <- as.numeric(np$psy_date - np$BIRTH) / 365.25


wmh <- wmh[, c("ID", "session_date", "WMH_volume")]
wmh$session_date <- as.Date(wmh$session_date, format = "%Y-%m-%d")
wmh <- merge(wmh, demogs, by = "ID", all = FALSE)
wmh$Age_at_Scan <- as.numeric(wmh$session_date - wmh$BIRTH) / 365.25


tau <- tau[, c("ID", "PET_Date", "Tauopathy")]
tau$PET_Date <- as.Date(tau$PET_Date, format = "%Y-%m-%d")
tau <- tau[!is.na(tau$Tauopathy),]
tau <- merge(tau, demogs, by = "ID", all = FALSE)
tau$Age_at_Scan <- as.numeric(tau$PET_Date - tau$BIRTH) / 365.25

mri <- mri[mri$FS_QC_Status == "Passed" | mri$FS_QC_Status == "Passed with edits",]
mri <- mri[,c("ID", "session_date", "CortSig_Thickness", "MR_TOTV_INTRACRANIAL", 
              "MR_LV_HIPPOCAMPUS", "MR_RV_HIPPOCAMPUS")]
mri$session_date <- as.Date(mri$session_date, format = "%Y-%m-%d")
mri <- merge(mri, demogs, by = "ID", all = FALSE)
mri$Age_at_Scan <- as.numeric(mri$session_date - mri$BIRTH) / 365.25

#Stuff for the tableone table
csf_comparison <- get_early_late(csf, "CSF_LP_DATE")
mmse_comparison <- get_early_late(mmse, "testdate")
mri_comparison <- get_early_late(mri, "session_date")
np_comparison <- get_early_late(np, "psy_date")
pet_comparison <- get_early_late(pet, "PET_Date")
tau_comparison <- get_early_late(tau, "PET_Date")
wmh_comparison <- get_early_late(wmh, "session_date")

df_tableone <- merge(csf_comparison, mmse_comparison[, c("ID", "cdr_baseline", "MMSE_baseline", "sumbox_baseline",
                                                         "cdr_recent", "MMSE_recent", "sumbox_recent")], by = "ID", all = TRUE)
df_tableone <- merge(df_tableone, mri_comparison[, c("ID", "CortSig_Thickness_baseline", "CortSig_Thickness_recent")], by = "ID", all = TRUE)
df_tableone <- merge(df_tableone, np_comparison[, c("ID", "lmdelay_baseline", "digsym_baseline", "lmdelay_recent", "digsym_recent")], by = "ID", all = TRUE)
df_tableone <- merge(df_tableone, pet_comparison[, c("ID", "pib_fsuvr_rsf_tot_cortmean_baseline", "pib_fsuvr_rsf_tot_cortmean_recent")], by = "ID", all = TRUE)
df_tableone <- merge(df_tableone, tau_comparison[, c("ID", "Tauopathy_baseline", "Tauopathy_recent")], by = "ID", all = TRUE)
df_tableone <- merge(df_tableone, wmh_comparison[, c("ID", "WMH_volume_baseline", "WMH_volume_recent")], by = "ID", all = TRUE)

tmp <- df_tableone[,c("ID", "CSF_LP_DATE_baseline")]
mmse_tmp <- mmse[, c("ID", "testdate", "cdr", "MMSE", "sumbox")]

setDT(tmp)
setDT(mmse_tmp)

setkey(mmse_tmp, ID, testdate)[, CSF_LP_DATE_baseline := testdate]
nearest_np <- mmse_tmp[tmp, roll = 'nearest']
colnames(nearest_np)[ 3:5] <- c("cdr_baseline", "MMSE_baseline", "sumbox_baseline")

df_tableone <- df_tableone[, !(names(df_tableone) %in% c("cdr_baseline", "MMSE_baseline",
                                                         "sumbox_baseline"))]

df_tableone <- merge(df_tableone, nearest_np, by = c("ID"))

```

```{r}


library(tableone)
myVars = c( "Age_at_LP_baseline", "Age_at_LP_recent", "GENDER_baseline", 
          "apoe_baseline", "race2_baseline", "EDUC_baseline",
           "cdr_baseline", "cdr_recent", 
          "MMSE_baseline", "MMSE_recent", 
          "sumbox_baseline", "sumbox_recent")
catVars = c( "GENDER_baseline", 
          "apoe_baseline", "race2_baseline",
           "cdr_baseline", "cdr_recent")
CreateTableOne(vars = myVars, strata = "upvoted_cluster_recent", data = df_tableone, factorVars = catVars)

tone_univariate <- tableone::CreateTableOne(vars = myVars, strata = "upvoted_cluster_recent", data = df_tableone, factorVars = catVars)
tone_univariate <- print(tone_univariate, printToggle = FALSE, noSpaces = TRUE)
library(kableExtra)
kable(tone_univariate, format = "pandoc")
```

### Individual trajectories vs Time

We expect AB42 to decline over time, so likely - given the orientation of the x axis in the figures above, timepoints are chronologically shown as you move from left to right for connected individuals. To validate this, we will plot the individual trajectories of participants with the most completed lumbar punctures in each of the three clusters. Our hypothesis is not exactly true, but it is generally true.

```{r}
csf_long <- merge(csf_long, cluster_results, by = "ID")
csf$ID <- as.factor(csf$ID)
csf_long$ID <- as.factor(csf_long$ID)
#c("Intermediate AD Biomarkers", "AD Biomarker Positive", "AD Biomarker Negative")
p1 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 5 & csf_long$upvoted_cluster == "AD Biomarker Negative", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw() + ylim(c(200, 2000))
p2 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 5 & csf_long$upvoted_cluster == "AD Biomarker Negative", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_pTau, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw() + ylim(c(20, 135))
p3 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 5 & csf_long$upvoted_cluster == "AD Biomarker Negative", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42_AB40, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw() + ylim(c(0.03, 0.12))
lay <- rbind(c(1, 2), c(3, 3))
lemon::grid_arrange_shared_legend(p1, p2, p3, nrow = 1, top = textGrob("Participants without Pathology and N = 6 Visits"), layout_matrix = lay)

```

```{r, eval=FALSE}
p1 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 5 & csf_long$upvoted_cluster == "Emerging", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw() + ylim(c(200, 2000))
p2 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 5 & csf_long$upvoted_cluster == "Emerging", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_pTau, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw()+ ylim(c(20, 135))
p3 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 5 & csf_long$upvoted_cluster == "Emerging", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42_AB40, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw()+ ylim(c(0.03, 0.12))
lemon::grid_arrange_shared_legend(p1, p2, p3, nrow = 1, top = textGrob("Participants with Emerging Pathology and N = 6 Visits"), layout_matrix = lay)

```

```{r, eval = FALSE}
p1 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 4 & csf_long$upvoted_cluster == "AD Biomarkers", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw() + ylim(c(200, 2000))
p2 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 4 & csf_long$upvoted_cluster == "AD Biomarkers", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_pTau, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw()+ ylim(c(20, 135))
p3 <- ggplot(csf[csf$ID %in% csf_long[csf_long$N > 4 & csf_long$upvoted_cluster == "AD Biomarkers", "ID"]$ID,],
       aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42_AB40, group = ID, colour = ID)) + geom_point() + geom_line() + theme_bw()+ ylim(c(0.03, 0.12))
lemon::grid_arrange_shared_legend(p1, p2, p3, nrow = 1, top = textGrob("Participants with AD Pathology and N = 5 or 6 Visits"), layout_matrix = lay)

csf$ID <- as.numeric(as.character(csf$ID)) #going back to structure we had before plotting
```



### Demographics

These are highly committed participants who have longitudinal measures for nearly every modality. We do not have much longitudinal PET Tau on these participants and thus in our summary table we only show a baseline tau pet value for each.

```{r}

table1( ~ Age_at_LP_baseline + Age_at_LP_recent + factor(GENDER_baseline) + 
          factor(apoe_baseline) + factor(race2_baseline) + EDUC_baseline +
          LUMIPULSE_CSF_AB42_baseline + LUMIPULSE_CSF_AB42_recent + 
          LUMIPULSE_CSF_AB40_baseline + LUMIPULSE_CSF_AB40_recent + 
          LUMIPULSE_CSF_AB42_AB40_baseline + LUMIPULSE_CSF_AB42_AB40_recent + 
          pib_fsuvr_rsf_tot_cortmean_baseline + pib_fsuvr_rsf_tot_cortmean_recent + 
          Tauopathy_baseline + 
          LUMIPULSE_CSF_pTau_baseline + LUMIPULSE_CSF_pTau_recent + 
          LUMIPULSE_CSF_tTau_baseline + LUMIPULSE_CSF_tTau_recent + 
          CortSig_Thickness_baseline + CortSig_Thickness_recent + 
          WMH_volume_baseline +WMH_volume_recent +
          CSF_NfL_baseline + CSF_NfL_recent + 
          factor(cdr_baseline) + factor(cdr_recent) + 
          MMSE_baseline + MMSE_recent + 
          sumbox_baseline + sumbox_recent + 
          lmdelay_baseline + lmdelay_recent + 
          digsym_baseline + digsym_recent | upvoted_cluster_baseline, data = df_tableone)

```

### Amyloid
Accepted CSF cutoffs for PET equivalency are shown as relevant


```{r}
pA1 <- ggplot(csf, aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42_AB40, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) + scale_colour_viridis(discrete = TRUE) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_hline(yintercept = 0.0673, linetype = "dashed", colour = "grey67") +
  xlab("Age at LP") + ylab("CSF AB42 / AB40 Ratio") + labs(colour = "Assigned Cluster")
pA2 <- ggplot(csf, aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB42, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) + scale_colour_viridis(discrete = TRUE) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() +
    xlab("Age at LP") + ylab("CSF AB42") + labs(colour = "Assigned Cluster")
pA3 <- ggplot(csf, aes(x = Age_at_LP, y = LUMIPULSE_CSF_AB40, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) + scale_colour_viridis(discrete = TRUE) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() +
    xlab("Age at LP") + ylab("CSF AB40") + labs(colour = "Assigned Cluster")
pA4 <- ggplot(pet, aes(x = Age_at_Scan, y = pib_fsuvr_rsf_tot_cortmean, group = ID, colour = upvoted_cluster)) + 
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + scale_colour_viridis(discrete = TRUE) +
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
    xlab("Age at LP") + ylab("PET PIB Summary Value") + labs(colour = "Assigned Cluster") +
  geom_hline(yintercept = 1.42, linetype = "dashed", colour = "grey67")

lemon::grid_arrange_shared_legend(pA1, pA2, pA3, pA4, nrow = 2, ncol = 2)



```

### Tau

Accepted CSF cutoffs for PET equivalency are shown as relevant

```{r}
pT1 <- ggplot(tau, aes(x = Age_at_Scan, y = Tauopathy, group = ID, colour = upvoted_cluster)) +
  scale_colour_viridis(discrete = TRUE) + labs(colour = "Assigned Cluster") +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "lm", aes(group = upvoted_cluster)) +
  geom_hline(yintercept = 1.22, linetype = "dashed", linecolour = "grey67") +
  xlab("Age at Scan") + ylab("PET AV1451 Summary Value")

pT2 <- ggplot(csf, aes(x = Age_at_LP, y = LUMIPULSE_CSF_pTau, group = ID, colour = upvoted_cluster)) + 
  scale_colour_viridis(discrete = TRUE) + labs(colour = "Assigned Cluster") +
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_hline(yintercept = 42.5, linetype = "dashed", linecolour = "grey67") +
  xlab("Age at LP") + ylab("CSF pTau")

lemon::grid_arrange_shared_legend(pT1, pT2,  nrow = 1, ncol = 2)


```

### Neurodegeneration

```{r}


pN4 <- ggplot(wmh, aes(x = Age_at_Scan, y = WMH_volume, group = ID, colour = upvoted_cluster)) +
  labs(colour = "Assigned Cluster") +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  scale_colour_viridis(discrete = TRUE) + xlab("Age at Scan") + ylab("WMH Volume") + scale_y_log10()

pN1 <- ggplot(mri, aes(x = Age_at_Scan, y = CortSig_Thickness, group = ID, colour = upvoted_cluster)) +
    labs(colour = "Assigned Cluster") +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
    scale_colour_viridis(discrete = TRUE) + xlab("Age at Scan") + ylab("Cortical Thickness")

p2 <- ggplot(mri, aes(x = Age_at_Scan, y = MR_LV_HIPPOCAMPUS, group = ID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))+
    scale_colour_viridis(discrete = TRUE)
p3 <- ggplot(mri, aes(x = Age_at_Scan, y = MR_RV_HIPPOCAMPUS, group = ID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))+
    scale_colour_viridis(discrete = TRUE)
lemon::grid_arrange_shared_legend(pN1, p2, p3, pN4, nrow = 2, ncol = 2)

pNFL1 <- ggplot(csf, aes(x = Age_at_LP, y = CSF_NfL, group = ID, colour = upvoted_cluster)) +
    labs(colour = "Assigned Cluster") +
  scale_colour_viridis(discrete = TRUE) + xlab("Age at LP") + ylab("CSF NfL") +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  scale_y_log10()
p2 <- ggplot(csf, aes(x = Age_at_LP, y = LN_CSF_NfL, group = ID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))+
    scale_colour_viridis(discrete = TRUE)
lemon::grid_arrange_shared_legend(pNFL1, p2,  nrow = 1, ncol = 2)

```

### Survival Analysis

We are able to complete a survival analysis on time to Amyloid Positivity (as measured by CSF AB42/AB40 ratio) and time to Tau Positivity (as measured by CSF pTau). When we do this, we omit the participants that are already amyloid positive (13 / 108) or tau positive (20 / 108) before study enrollment. We consider the age at pathology positivity to be the first CSF appointment with a value surpassing either CSF AB42/40 < 0.0673 or CSF pTau > 42.5. When we do this we see statistically significant differences across the three strata for both Amyloid and Tau. These differences are what we would expect. 

```{r}
library(survival)
library(ggfortify)

csf_surv <- read.csv(".././Data/csf_processed_for_analysis.csv")
csf_surv$upvoted_cluster <- factor(csf_surv$upvoted_cluster, levels = c("Emerging", "AD Pathology", "Pathology Free"))
levels(csf_surv$upvoted_cluster) <- c("Intermediate AD Biomarkers", "AD Biomarker Positive", "AD Biomarker Negative") #renaming to match dave holtzman convention 

csf_surv$Apos <- ifelse(csf_surv$LUMIPULSE_CSF_AB42_AB40 < 0.0673, 1, 0)
csf_surv$Tpos <- ifelse(csf_surv$LUMIPULSE_CSF_pTau > 42.5, 1, 0) 
csf_surv$CSF_LP_DATE <- as.Date(csf_surv$CSF_LP_DATE, format = "%Y-%m-%d")

csf_baseline_surv <- setDT(csf_surv)[order(CSF_LP_DATE), head(.SD, 1L), by = ID]
csf_final_surv<- setDT(csf_surv)[order(-CSF_LP_DATE), head(.SD, 1L), by = ID]

cdr_surv <- mmse[, c("ID", "testdate", "cdr")]
cdr_surv$testdate <- as.Date(cdr_surv$testdate, format = "%Y-%m-%d")
cdr_surv <- merge(cdr_surv, demogs[, c("ID", "BIRTH")], by = "ID")
cdr_surv$Age <- as.numeric(cdr_surv$testdate - as.Date(cdr_surv$BIRTH, format = "%Y-%m-%d")) / 365.25

cdr_baseline_surv <- MatchbyNearestDate(data.frame(csf_baseline_surv[, c("ID", "CSF_LP_DATE", "upvoted_cluster")]),data.frame(cdr_surv), 
                                        "ID", "CSF_LP_DATE", "testdate")

cdr_final <- cdr_surv[cdr_surv$cdr > 0,]
cdr_final <- setDT(cdr_final)[order(testdate), head(.SD, 1L), by = ID] #getting first nonzero cdr rating
cdr_final <- cdr_final[cdr_final$ID %in% csf_final_surv$ID,]

cdr_latest <- setDT(cdr_surv[cdr_surv$ID %in% csf_final_surv$ID,])[order(-testdate), head(.SD, 1L), by = ID] 

cdr_final_surv <- rbind(cdr_final, cdr_latest[!(cdr_latest$ID %in% cdr_final$ID),])
cdr_final_surv <- merge(cdr_final_surv, cdr_baseline_surv[, c("ID", "upvoted_cluster")], by = "ID")
rm(cdr_final, cdr_latest)

cdr_time_to_pos <- rbind(cdr_baseline_surv[, names(cdr_final_surv)], cdr_final_surv)

#13 / 108 become apos during study
#Dropping amyloid positives before study starts
csf_time_to_amyloid <- csf_surv[!(csf_surv$ID %in% csf_baseline_surv[csf_baseline_surv$Apos == 1, "ID"]$ID),]
csf_time_to_amyloid <- csf_time_to_amyloid[, c("ID", "Age_at_LP", "Apos", "upvoted_cluster")]

km_trt_fit_A <- survfit(Surv(Age_at_LP, Apos) ~ upvoted_cluster, data=csf_time_to_amyloid)
p1 <- autoplot(km_trt_fit_A) + theme_bw() + xlab("Age") + ylab("Percent Amyloid Negative") +
  xlim(c(47, 87))+
    scale_colour_viridis(discrete = TRUE) + scale_fill_viridis(discrete = TRUE) +
  geom_hline(yintercept = 0.5, linetype = "dashed")

fit_A <- coxph(Surv(Age_at_LP, Apos) ~ upvoted_cluster, data = csf_time_to_amyloid)
model_A <- anova(fit_A)
summary(model_A)



#20 / 108 become tpos during study
csf_time_to_tau <- csf_surv[!(csf_surv$ID %in% csf_baseline_surv[csf_baseline_surv$Tpos == 1, "ID"]$ID),]
csf_time_to_tau <- csf_time_to_tau[, c("ID", "Age_at_LP", "Tpos", "upvoted_cluster")]

csf_time_to_tau_Tpos <- setDT(csf_time_to_tau[csf_time_to_tau$Tpos == 1,])[order(Age_at_LP), head(.SD, 1L), by = ID]
csf_time_to_tau_Tneg <- csf_time_to_tau[!(csf_time_to_tau$ID %in% csf_time_to_tau_Tpos$ID),]
csf_time_to_tau_Tneg <- setDT(csf_time_to_tau_Tneg)[order(Age_at_LP), tail(.SD, 1L), by = ID]

csf_time_to_tau <- rbind(csf_time_to_tau_Tpos, csf_time_to_tau_Tneg)


km_trt_fit_T <- survfit(Surv(Age_at_LP, Tpos) ~ upvoted_cluster, data=csf_time_to_tau)
p2 <- autoplot(km_trt_fit_T) + theme_bw() + xlab("Age") + ylab("Percent Tau Negative")+
  xlim(c(47, 87))+
    scale_colour_viridis(discrete = TRUE)  + scale_fill_viridis(discrete = TRUE) +
  geom_hline(yintercept = 0.5, linetype = "dashed")

fit_T <- coxph(Surv(Age_at_LP, Tpos) ~ upvoted_cluster, data = csf_time_to_tau)
model_T <- anova(fit_T)
summary(model_T)


cdr_time_to_pos$cdr <- ifelse(cdr_time_to_pos$cdr > 0, 1, 0)
km_trt_fit_cdr <- survfit(Surv(Age, cdr) ~ upvoted_cluster, data=cdr_time_to_pos)
p3 <- autoplot(km_trt_fit_cdr) + theme_bw() + xlab("Age") + ylab("Percent CDR > 0") +
  xlim(c(47, 87))+
    scale_colour_viridis(discrete = TRUE) + scale_fill_viridis(discrete = TRUE) +
  geom_hline(yintercept = 0.5, linetype = "dashed")


g <- lemon::grid_arrange_shared_legend(p1, p2, p3, 
                                  nrow = 3, ncol = 1, top = "Survival Analysis - Time to Pathology / Symptoms")

ggsave(g, filename = 'Fig2.png', dpi = 300, type = 'cairo',
       width = 8, height = 10, units = 'in')
```

```{r}
survival_df_T <- data.frame("time" = as.vector(km_trt_fit_T$time),
                            "surv" = as.vector(km_trt_fit_T$surv),
                            "cumhaz" = as.vector(km_trt_fit_T$cumhaz),
                            "std.err" = as.vector(km_trt_fit_T$std.err),
                            "Cluster" = c(rep("Intermediate AD Biomarkers", 18), rep("AD Biomarker Positive", 3), rep("AD Biomarker Negative", 61)))

```

```{r}
#Trying to figure out how to plot both A's on the same curve and both T's on the same curve
csf_time_to_amyloid$Pathology <- "Amyloid"
csf_time_to_tau$Pathology <- "Tau"
colnames(csf_time_to_amyloid)[3] <- "Positive"
colnames(csf_time_to_tau)[3] <- "Positive"

csf_time_to_pathology_emerging <- rbind(csf_time_to_amyloid[csf_time_to_amyloid$upvoted_cluster == "Intermediate AD Biomarkers"],
                                        csf_time_to_tau[csf_time_to_tau$upvoted_cluster == "Intermediate AD Biomarkers",])

km_trt_fit_emerging <- survfit(Surv(Age_at_LP, Positive) ~ Pathology, data = csf_time_to_pathology_emerging)
p3 <- autoplot(km_trt_fit_emerging) + theme_bw() + xlab("Age") + ylab("Percent Pathology Negative")+
  xlim(c(45, 90))+
    scale_colour_viridis(discrete = TRUE)  + scale_fill_viridis(discrete = TRUE) + ggtitle("Intermediate AD Biomarker Cluster")


csf_time_to_pathology_AD <- rbind(csf_time_to_amyloid[csf_time_to_amyloid$upvoted_cluster == "AD Biomarker Positive"],
                                        csf_time_to_tau[csf_time_to_tau$upvoted_cluster == "AD Biomarker Positive",])

km_trt_fit_AD <- survfit(Surv(Age_at_LP, Positive) ~ Pathology, data = csf_time_to_pathology_AD)
p4 <- autoplot(km_trt_fit_AD) + theme_bw() + xlab("Age") + ylab("Percent Pathology Negative")+
  xlim(c(45, 90))+
    scale_colour_viridis(discrete = TRUE)  + scale_fill_viridis(discrete = TRUE) + ggtitle("AD Biomarker Positive Cluster")

lay <- rbind(c(1, 1, 3), c(2, 2, 4))
grid.arrange(p1, p2, p3, p4, layout_matrix = lay)

```


We also have proteomics data on a subset of these participants. In a somewhat shocking turn of events, we can reliably distinguish - via pairwise supervised clustering methods - between individuals classified as "emerging" vs. those classified as having AD pathology, between individuals classified as emerging vs those having no pathology, and between individuals classified as having AD pathology and those who have no pathology.

Of note - this is not the best way to talk about proteomics. Pelora generates lists of clusters working in tandem. I'm really not sure how to think about them - we need to consult with Carlos. Further, because I used cross-validation (which I think is appropriate given the N), how do we balance each independent model output? I need advice.

```{r}
#Note....these results are generated via the proteomicsCheck.R script
df <- readRDS(".././Results/SuccessfulProteomicsClassificationofADvsNonAD_genes.RDS")
df <- data.frame("Genes" = unlist(df))
df <- data.frame(table(unlist(df)))
df <- df[with(df, order(-Freq)),][1:10,]

auc <- readRDS(".././Results/SuccessfulProteomicsClassificationofADvsNonAD_auc.RDS")


ggplot(df, aes(x = reorder(Var1,  Freq), y = Freq)) + geom_col() + 
  coord_flip() + theme_bw() + xlab("Gene") + 
  labs(title = "AD vs Non-AD Classification",
       subtitle = "Most Frequently Utilized Genes in Pelora Classification",
       caption = "10 fold cross validation") + theme(axis.text.y = element_text(angle = 45)) +
  annotate("text", x = 1, y = 1.5 * mean(df$Freq), label = paste0("AUC: ", round(mean(unlist(auc)), 3)),
           size = 4)

```

```{r}
df <- readRDS(".././Results/SuccessfulProteomicsClassificationofEmergingvsAD_genes.RDS")
df <- data.frame("Genes" = unlist(df))
df <- data.frame(table(unlist(df)))
df <- df[with(df, order(-Freq)),][1:10,]

auc <- readRDS(".././Results/SuccessfulProteomicsClassificationofEmergingvsAD_auc.RDS")


ggplot(df, aes(x = reorder(Var1,  Freq), y = Freq)) + geom_col() + 
  coord_flip() + theme_bw() + xlab("Gene") + 
  labs(title = "AD vs Emerging Pathology Classification",
       subtitle = "Most Frequently Utilized Genes in Pelora Classification",
       caption = "10 fold cross validation") + theme(axis.text.y = element_text(angle = 45)) +
  annotate("text", x = 1, y = 1.25 * mean(df$Freq), label = paste0("AUC: ", round(mean(unlist(auc)), 3)),
           size = 4)

```

```{r}
df <- readRDS(".././Results/SuccessfulProteomicsClassificationofEmergingvsNonAD_genes.RDS")
df <- data.frame("Genes" = unlist(df))
df <- data.frame(table(unlist(df)))
df <- df[with(df, order(-Freq)),][1:10,]

auc <- readRDS(".././Results/SuccessfulProteomicsClassificationofEmergingvsNonAD_auc.RDS")


ggplot(df, aes(x = reorder(Var1,  Freq), y = Freq)) + geom_col() + 
  coord_flip() + theme_bw() + xlab("Gene") + 
  labs(title = "No Pathology vs Emerging Pathology Classification",
       subtitle = "Most Frequently Utilized Genes in Pelora Classification",
       caption = "10 fold cross validation") + theme(axis.text.y = element_text(angle = 45)) +
  annotate("text", x = 1, y = 1.5 * mean(df$Freq), label = paste0("AUC: ", round(mean(unlist(auc)), 3)),
           size = 4)

```


I'm thinking that using cross-validation helps confirm for us that there is reasonable robustness to the model, but when we want to actually make inferences for this cohort, we should train on the entire dataset and examine those gene clusters.

#### AD vs Non-AD

When we train and test on the same dataset, we do better than the cross-validated number, which is to be expected.

```{r, AD_NOneSingleRun, cache = TRUE}
IDs <- read.csv(".././Data/Clustered_Participants.csv")
csf_df <- read.csv(".././Data/20190925_CSF_afterQC_proteomic_exprs.csv")

#Reading in proteomic datasets
df_pro_csf <- read.csv(".././data/v4_CSF_afterQC_BasicPheno-All.csv")
df_pro_csf$MAPID <- as.numeric(as.character(df_pro_csf$MAPID))
df_pro_csf <- df_pro_csf[, c("MAPID", "ExtIdentifier", "TimePoint",  "CDR.Score")]


df_pro_csf$TimePoint <-as.Date(format(as.Date(as.character(df_pro_csf$TimePoint), "%m/%d/%y"), "20%y-%m-%d"))


csf_df <- gather(csf_df, ex_ID,measurement,  EXID40000001884654:EXID40000001735507)

csf_df <- merge(csf_df, df_pro_csf, by.x = "ex_ID", by.y = "ExtIdentifier", all = FALSE)

csf_df <- merge(csf_df, IDs, by.x = "MAPID", by.y = "ID", all = FALSE)

csf_check <- IDs[!duplicated(IDs[, c("ID", "upvoted_cluster")]), c("ID", "upvoted_cluster")]


csf_df <- csf_df[!duplicated(csf_df),]
csf_df <- data.table(csf_df)[, measurement := mean(measurement, na.rm = TRUE), by = list(MAPID, TargetFullName, TimePoint)]
csf_df <- unique(csf_df)
csf_df <- spread(csf_df[, c("MAPID", "TargetFullName", "measurement", "TimePoint")], TargetFullName, measurement)
csf_df <- unique(data.table(csf_df)[order(-TimePoint)], by="MAPID")
csf_df <- merge(csf_df, csf_check, by.x = "MAPID", by.y = "ID", all = FALSE)

#median imputation for missingness
f=function(x){
  x<-as.numeric(as.character(x)) #first convert each column into numeric if it is from factor
  x[is.na(x)] =median(x, na.rm=TRUE) #convert the item with NA to median value from the column
  x #display the column
}
imputed_medians=data.frame(apply(csf_df,2,f))

csf_df <- cbind(csf_df[,1:2], imputed_medians[,3:715], csf_df[,716])
colnames(csf_df)[716] <- "upvoted_cluster"

csf_df$upvoted_cluster <- as.factor(csf_df$upvoted_cluster)


# csf_df$upvoted_cluster <- as.numeric(as.character(csf_df$upvoted_cluster))
# csf_df <- csf_df[!is.na(csf_df$upvoted_cluster),]
NumClust <- 10

#need to run below for each comparison
csf_analysis <- csf_df
levels(csf_analysis$upvoted_cluster) <- c(1, NA, 0) #classifying both ad and emerging as 1's, healthy as 0's
csf_analysis <- csf_analysis[!is.na(csf_analysis$upvoted_cluster)]
#csf_analysis <- csf_analysis[with(csf_analysis, order(upvoted_cluster))]


set.seed(6646)
  mod1 <- pelora(as.matrix(csf_analysis[,3:715]), as.numeric(as.character(csf_analysis$upvoted_cluster)), noc = NumClust,
                 standardize = TRUE)

    tmp <- data.frame(mod1$values)
  tmp$Classification <- as.factor(csf_analysis$upvoted_cluster)
  X_names <- paste0("X", seq(from = 1, to = NumClust))
  #mod_tmp <- glm(as.formula(paste("Classification ~ ", paste(X_names, collapse= "+"))), family = "binomial", data = tmp)
  
  cvfit <- cv.glmnet(as.matrix(tmp[, X_names]), tmp$Classification, family = "binomial", type.measure = "class", nfolds = 10)
  # plot(cvfit)
  retained_nums <- data.frame(as.matrix(coef(cvfit, s = "lambda.min")))
  retained_nums$keeps <- row.names(retained_nums)
  retained_nums <- retained_nums[!(retained_nums$X1 == 0), "keeps"]
  retained_nums <- retained_nums[2:length(retained_nums)]
  retained_nums <- tidyr::extract_numeric(retained_nums)
  
  test_matrix <- predict(mod1, csf_analysis[,3:715])
  colnames(test_matrix) <- X_names
  
  standardized_test_matrix <- standardize.genes(test_matrix[, X_names])
  
  results <- data.frame("Probability" = predict(cvfit, newx = standardized_test_matrix$x, s = "lambda.min", type = "response"),
                        "Reality" = csf_analysis$upvoted_cluster)
  results$Prediction <- ifelse(results$X1 < 0.5, 1, 0)
    roc_obj <- roc(results$Reality, results$X1)
    
    plot(roc_obj)
    
    print(auc(roc_obj))
    
    tmp_list <- list()
    crit_list <- list()
    vec_list <- list()

  for(j in 1:length(retained_nums)){
    tmp_list[[j]] <- mod1$gene.names[mod1$genes[[retained_nums[j]]]]
    crit_list[[j]] <- mod1$crit[[retained_nums[j]]]
    vec_list[[j]] <- rep(retained_nums[j], length(mod1$crit[[retained_nums[j]]]))}
    
    print("First Cluster of Proteomics used for classification:")
    tmp_list[[1]]
    
    print("Second Cluster of Proteomics used for classification:")
    tmp_list[[2]]
    
    df_AD_no_crit <- data.frame("Gene" = unlist(tmp_list),
                                      "Crit" = unlist(crit_list),
                                      "Group" = unlist(vec_list))
    
    print("First Cluster of Proteomics used for classification:")
    tmp_list[[1]]
    
    print("Second Cluster of Proteomics used for classification:")
    tmp_list[[2]]
    
    print("Third Cluster of Proteomics used for classification:")
    tmp_list[[3]]
```

Here we visualize the proteins that are up and down regulated.

```{r, fig.height = 40, fig.width = 30}
protein_names <- unlist(tmp_list)
protein_names <- c( unique(protein_names))
protein_names <- gsub(" ", ".", protein_names)
protein_names <- gsub("-", ".", protein_names)
protein_names <- gsub("/", ".", protein_names)
protein_names <- gsub(",", ".", protein_names)
protein_names <- gsub("(", ".", protein_names, fixed = TRUE)
protein_names <- gsub(")", ".", protein_names)
protein_names <- gsub(":", ".", protein_names)
protein_names <- gsub("[", ".", protein_names, fixed = TRUE)
protein_names <- gsub("]", ".", protein_names, fixed = TRUE)
protein_names <- gsub("+", ".", protein_names, fixed = TRUE)



csf_analysis <- data.frame(csf_analysis)
colnames(csf_analysis)[3:8] <- c("14.3.3.protein.beta.alpha", "14.3.3.protein.epsilon", "14.3.3.protein.family",    
                           "14.3.3.protein.sigma",  "14.3.3.protein.theta","14.3.3.protein.zeta.delta")
colnames(csf_analysis)[13:14] <- c("60.kDa.heat.shock.protein..mitochondrial", "72.kDa.type.IV.collagenase")
# 
# heatmap_df <-as.matrix(csf_analysis[, protein_names])
# heatmap_df_scaled <- normalize(heatmap_df)

levels(csf_analysis$upvoted_cluster) <- c("AD", "No Path")
#https://cran.r-project.org/web/packages/heatmaply/vignettes/heatmaply.html
bp1 <- get_barplot( df_AD_no_crit) + ggtitle("AD vs. No Pathology", subtitle = "Protein Clusters Relevant to Classification")


```



#### AD vs Emerging Pathology 

```{r, ADEmergingSingleRun, cache = TRUE}
#need to run below for each comparison
csf_analysis <- csf_df
levels(csf_analysis$upvoted_cluster) <- c(1, 0, NA) #classifying both ad and emerging as 1's, healthy as 0's
csf_analysis <- csf_analysis[!is.na(csf_analysis$upvoted_cluster)]
csf_analysis <- csf_analysis[with(csf_analysis, order(upvoted_cluster))]


set.seed(6646)
  mod1 <- pelora(as.matrix(csf_analysis[,3:715]), as.numeric(as.character(csf_analysis$upvoted_cluster)), noc = NumClust,
                 standardize = TRUE)
  
    tmp <- data.frame(mod1$values)
  tmp$Classification <- as.factor(csf_analysis$upvoted_cluster)
  X_names <- paste0("X", seq(from = 1, to = NumClust))
  #mod_tmp <- glm(as.formula(paste("Classification ~ ", paste(X_names, collapse= "+"))), family = "binomial", data = tmp)
  
  cvfit <- cv.glmnet(as.matrix(tmp[, X_names]), tmp$Classification, family = "binomial", type.measure = "class", nfolds = 10)
  # plot(cvfit)
  retained_nums <- data.frame(as.matrix(coef(cvfit, s = "lambda.min")))
  retained_nums$keeps <- row.names(retained_nums)
  retained_nums <- retained_nums[!(retained_nums$X1 == 0), "keeps"]
  retained_nums <- retained_nums[2:length(retained_nums)]
  retained_nums <- tidyr::extract_numeric(retained_nums)
  
  test_matrix <- predict(mod1, csf_analysis[,3:715])
  colnames(test_matrix) <- X_names
  
  standardized_test_matrix <- standardize.genes(test_matrix[, X_names])
  
  results <- data.frame("Probability" = predict(cvfit, newx = standardized_test_matrix$x, s = "lambda.min", type = "response"),
                        "Reality" = csf_analysis$upvoted_cluster)
  results$Prediction <- ifelse(results$X1 < 0.5, 1, 0)
    roc_obj <- roc(results$Reality, results$X1)
    
    plot(roc_obj)
    
    print(auc(roc_obj))
    
    tmp_list <- list()
    crit_list <- list()
    vec_list <- list()

  for(j in 1:length(retained_nums)){
    tmp_list[[j]] <- mod1$gene.names[mod1$genes[[retained_nums[j]]]]
    crit_list[[j]] <- mod1$crit[[retained_nums[j]]]
    vec_list[[j]] <- rep(retained_nums[j], length(mod1$crit[[retained_nums[j]]]))}
    
    print("First Cluster of Proteomics used for classification:")
    tmp_list[[1]]
    
    print("Second Cluster of Proteomics used for classification:")
    tmp_list[[2]]
    
    df_AD_emerging_crit <- data.frame("Gene" = unlist(tmp_list),
                                      "Crit" = unlist(crit_list),
                                      "Group" = unlist(vec_list))

```

```{r, fig.height = 40, fig.width = 30}
protein_names <- unlist(tmp_list)
protein_names <- c( unique(protein_names))
protein_names <- gsub(" ", ".", protein_names)
protein_names <- gsub("-", ".", protein_names)
protein_names <- gsub("/", ".", protein_names)
protein_names <- gsub(",", ".", protein_names)
protein_names <- gsub("(", ".", protein_names, fixed = TRUE)
protein_names <- gsub(")", ".", protein_names)
protein_names <- gsub(":", ".", protein_names)
protein_names <- gsub("[", ".", protein_names, fixed = TRUE)
protein_names <- gsub("]", ".", protein_names, fixed = TRUE)
protein_names <- gsub("+", ".", protein_names, fixed = TRUE)



csf_analysis <- data.frame(csf_analysis)
colnames(csf_analysis)[3:8] <- c("14.3.3.protein.beta.alpha", "14.3.3.protein.epsilon", "14.3.3.protein.family",    
                           "14.3.3.protein.sigma",  "14.3.3.protein.theta","14.3.3.protein.zeta.delta")
colnames(csf_analysis)[13:14] <- c("60.kDa.heat.shock.protein..mitochondrial", "72.kDa.type.IV.collagenase")


# heatmap_df <-as.matrix(csf_analysis[, protein_names])
# heatmap_df_scaled <- normalize(heatmap_df)

levels(csf_analysis$upvoted_cluster) <- c("AD", "Emerging")
#https://cran.r-project.org/web/packages/heatmaply/vignettes/heatmaply.html

bp2 <-  get_barplot(df_AD_emerging_crit) + ggtitle("AD vs. Emerging", subtitle = "Protein Clusters Relevant to Classification")



```

#### Emerging Pathology vs No Pathology

```{r, Emerging_NOneSingleRun, cache = TRUE}
#need to run below for each comparison
csf_analysis <- csf_df
levels(csf_analysis$upvoted_cluster) <- c(NA, 1, 0) #classifying both ad and emerging as 1's, healthy as 0's
csf_analysis <- csf_analysis[!is.na(csf_analysis$upvoted_cluster)]
csf_analysis <- csf_analysis[with(csf_analysis, order(upvoted_cluster))]


set.seed(7777)
  mod1 <- pelora(as.matrix(csf_analysis[,3:715]), as.numeric(as.character(csf_analysis$upvoted_cluster)), noc = NumClust,
                 standardize = TRUE)
  
    tmp <- data.frame(mod1$values)
  tmp$Classification <- as.factor(csf_analysis$upvoted_cluster)
  X_names <- paste0("X", seq(from = 1, to = NumClust))
  #mod_tmp <- glm(as.formula(paste("Classification ~ ", paste(X_names, collapse= "+"))), family = "binomial", data = tmp)
  
  cvfit <- cv.glmnet(as.matrix(tmp[, X_names]), tmp$Classification, family = "binomial", type.measure = "class", nfolds = 10)
  # plot(cvfit)
  retained_nums <- data.frame(as.matrix(coef(cvfit, s = "lambda.min")))
  retained_nums$keeps <- row.names(retained_nums)
  retained_nums <- retained_nums[!(retained_nums$X1 == 0), "keeps"]
  retained_nums <- retained_nums[2:length(retained_nums)]
  retained_nums <- tidyr::extract_numeric(retained_nums)
  
  test_matrix <- predict(mod1, csf_analysis[,3:715])
  colnames(test_matrix) <- X_names
  
  standardized_test_matrix <- standardize.genes(test_matrix[, X_names])
  
  results <- data.frame("Probability" = predict(cvfit, newx = standardized_test_matrix$x, s = "lambda.min", type = "response"),
                        "Reality" = csf_analysis$upvoted_cluster)
  results$Prediction <- ifelse(results$X1 < 0.5, 1, 0)
    roc_obj <- roc(results$Reality, results$X1)
    
    plot(roc_obj)
    
    print(auc(roc_obj))
    
    tmp_list <- list()
    crit_list <- list()
    vec_list <- list()

  for(j in 1:length(retained_nums)){
    tmp_list[[j]] <- mod1$gene.names[mod1$genes[[retained_nums[j]]]]
    crit_list[[j]] <- mod1$crit[[retained_nums[j]]]
    vec_list[[j]] <- rep(retained_nums[j], length(mod1$crit[[retained_nums[j]]]))}
    
    print("First Cluster of Proteomics used for classification:")
    tmp_list[[1]]
    
    print("Second Cluster of Proteomics used for classification:")
    tmp_list[[2]]
    
    df_emerging_no_crit <- data.frame("Gene" = unlist(tmp_list),
                                      "Crit" = unlist(crit_list),
                                      "Group" = unlist(vec_list))
    
    print("First Cluster of Proteomics used for classification:")
    tmp_list[[1]]
    
    print("Second Cluster of Proteomics used for classification:")
    tmp_list[[2]]
    
    print("Third Cluster of Proteomics used for classification:")
    tmp_list[[3]]
    
        print("Fourth Cluster of Proteomics used for classification:")
    tmp_list[[4]]
    
        print("Fifth Cluster of Proteomics used for classification:")
    tmp_list[[5]]
    
        print("Sixth Cluster of Proteomics used for classification:")
    tmp_list[[6]]
    
        print("Seventh Cluster of Proteomics used for classification:")
    tmp_list[[7]]
    
        print("Eighth Cluster of Proteomics used for classification:")
    tmp_list[[8]]
    

```

```{r, fig.height = 40, fig.width = 30}
protein_names <- unlist(tmp_list)
protein_names <- c( unique(protein_names))
protein_names <- gsub(" ", ".", protein_names)
protein_names <- gsub("-", ".", protein_names)
protein_names <- gsub("/", ".", protein_names)
protein_names <- gsub(",", ".", protein_names)
protein_names <- gsub("(", ".", protein_names, fixed = TRUE)
protein_names <- gsub(")", ".", protein_names)
protein_names <- gsub(":", ".", protein_names)
protein_names <- gsub("[", ".", protein_names, fixed = TRUE)
protein_names <- gsub("]", ".", protein_names, fixed = TRUE)
protein_names <- gsub("+", ".", protein_names, fixed = TRUE)



csf_analysis <- data.frame(csf_analysis)
colnames(csf_analysis)[3:8] <- c("14.3.3.protein.beta.alpha", "14.3.3.protein.epsilon", "14.3.3.protein.family",    
                           "14.3.3.protein.sigma",  "14.3.3.protein.theta","14.3.3.protein.zeta.delta")
colnames(csf_analysis)[12:14] <- c("6.phosphogluconate.dehydrogenase..decarboxylating",
                                   "60.kDa.heat.shock.protein..mitochondrial", "72.kDa.type.IV.collagenase")

#protein_names[!(protein_names %in% names(csf_analysis))]
# heatmap_df <- csf_analysis[, protein_names]
# heatmap_df <-as.matrix(heatmap_df)
# heatmap_df_scaled <- normalize(heatmap_df)

levels(csf_analysis$upvoted_cluster) <- c("Emerging", "No Path")
#https://cran.r-project.org/web/packages/heatmaply/vignettes/heatmaply.html
bp3 <-  get_barplot(df_emerging_no_crit) + ggtitle("Emerging vs. no Pathology", subtitle = "Protein Clusters Relevant to Classification")

```

```{r, fig.height = 40, fig.width = 30}
grid.arrange(bp1, bp2, bp3, nrow = 1)
```
#### Further Analysis of Proteins of Interest

Liver Related Proteins:
```{r}


csf_posthoc <- merge(csf_df, csf[,c("ID", "CSF_LP_DATE", "LUMIPULSE_CSF_AB42", "LUMIPULSE_CSF_pTau",
                                          "LUMIPULSE_CSF_AB42_AB40", "Age_at_LP")], by.x = c("MAPID", "TimePoint"),
                     by.y = c("ID", "CSF_LP_DATE"), all.x = TRUE, all.y = FALSE)

my_comparisons <- list( c("AD Biomarker Positive", "AD Biomarker Negative"), 
                        c("AD Biomarker Positive", "Intermediate AD Biomarkers"), 
                        c("Intermediate AD Biomarkers", "AD Biomarker Negative") )

p1 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Retinol-binding protein 4",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p2 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Hemopexin",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p3 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Tumor necrosis factor-inducible gene 6 protein",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


lemon::grid_arrange_shared_legend(p1, p2, p3, nrow = 1, ncol = 3)
```

BBB Proteins:

```{r}
p1 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "SPARC-related modular calcium-binding protein 1",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p2 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Nidogen-2",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p3 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Polymeric immunoglobulin receptor",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p4 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Angiogenin",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


lemon::grid_arrange_shared_legend(p1, p2, p3, p4, nrow = 2, ncol = 2)

```



Protective Proteins:

```{r}
p1 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Tyrosine-protein kinase receptor TYRO3",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p2 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Insulin-like growth factor-binding protein 1",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p3 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Haptoglobin",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p4 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Insulin-like growth factor-binding protein 3",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p5 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Complement C1r subcomponent",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p6 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Pituitary adenylate cyclase-activating polypeptide 27",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


lemon::grid_arrange_shared_legend(p1, p2, p3, p4, p5, p6, nrow = 2, ncol = 3)

```
Calcineurin makes up about 1% of total neural proteins. It is involved in neuronal plasticity and has been shown to be hyper-active within AD (Reese & Taglialatela, 2011).

```{r}

ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Calcineurin",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


p1 <- ggplot(csf_posthoc, aes(x = Age_at_LP, y = Calcineurin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p2 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_pTau, y = Calcineurin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_smooth(method = "gam", aes(group = 1), colour = "black")
p3 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42, y = Calcineurin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p4 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42_AB40, y = Calcineurin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) 
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)

```

SPARC enhances Blood Brain Barrier permeability, promotes neuroinflammation, and prolongs pro-inflammatory M1 phase of microglia (Pilozzi et al, 2020).

```{r}
ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "SPARC-related modular calcium-binding protein 1",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


p1 <- ggplot(csf_posthoc, aes(x = Age_at_LP, y = `SPARC-related modular calcium-binding protein 1`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p2 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_pTau, y = `SPARC-related modular calcium-binding protein 1`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p3 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42, y = `SPARC-related modular calcium-binding protein 1`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p4 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42_AB40, y = `SPARC-related modular calcium-binding protein 1`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) 
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)

```

The family of 14-3-3 proteins are scaffolding proteins that interact with tau and regulate tau phosphorylation and aggregation (Chen et al, 2019)

```{r}
p1 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "14-3-3 protein beta/alpha",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p2 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "14-3-3 protein epsilon",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p3 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "14-3-3 protein family",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p4 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "14-3-3 protein sigma",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p5 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "14-3-3 protein theta",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)
p6 <- ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "14-3-3 protein zeta/delta",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)

lemon::grid_arrange_shared_legend(p1, p2, p3, p4, p5, p6, nrow = 2, ncol = 3)


```

Angiogen is most commonly associated with Parksinsons and ALS, but a mutation in Angiogen has also been found in association with AD (Stella et al, 2019). It is expressed and enriched in motor neurons, and accumulates under conditions of cellular stress like oxidative stress (Prehn & Jirstrom, 2020).

WOW. Look at the elevated angiogen for the AD people with low pTau levels. Story about early oxidative stress leading to accelerated pace towards AD?

```{r}
ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Angiogenin",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


p1 <- ggplot(csf_posthoc, aes(x = Age_at_LP, y = Angiogenin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p2 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_pTau, y = Angiogenin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p3 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42, y = Angiogenin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p4 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42_AB40, y = Angiogenin, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) 
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)

```

Inhibition of galectin-8 increases seeded tau aggregation.

```{r}
ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Galectin-8",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


p1 <- ggplot(csf_posthoc, aes(x = Age_at_LP, y = `Galectin-8`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p2 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_pTau, y = `Galectin-8`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p3 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42, y = `Galectin-8`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p4 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42_AB40, y = `Galectin-8`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) 
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)

```

Glycerol-3-phosphate dehydrogenase [NAD(+)], cytoplasmic was originally identified as involved in aerobic metabolism of glucose. It has been found to associate with APP and is significantly inhibited in the progression of AD due to oxidative modification (Butterfield et al, 2010).

```{r}
ggpubr::ggboxplot(csf_posthoc, x = "upvoted_cluster", y = "Glycerol-3-phosphate dehydrogenase [NAD(+)], cytoplasmic",
          color = "upvoted_cluster")+ 
  ggpubr::stat_compare_means(comparisons = my_comparisons)


p1 <- ggplot(csf_posthoc, aes(x = Age_at_LP, y = `Glycerol-3-phosphate dehydrogenase [NAD(+)], cytoplasmic`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p2 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_pTau, y = `Glycerol-3-phosphate dehydrogenase [NAD(+)], cytoplasmic`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p3 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42, y = `Glycerol-3-phosphate dehydrogenase [NAD(+)], cytoplasmic`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p4 <- ggplot(csf_posthoc, aes(x = LUMIPULSE_CSF_AB42_AB40, y = `Glycerol-3-phosphate dehydrogenase [NAD(+)], cytoplasmic`, group = MAPID, colour = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster)) 
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)

```

### Cognition

lmdelay seems to be a very sensitive cognitive test

```{r}
p1 <- ggplot(mmse, aes(x = Age_at_Visit, y = MMSE, group = ID, colour = upvoted_cluster)) + 
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_hline(yintercept = 23, linetype = "dashed", colour = "red") #cutoff for dementia
p2 <- ggplot(mmse, aes(x = Age_at_Visit, y = sumbox, group = ID, colour = upvoted_cluster)) +
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p3 <- ggplot(mmse, aes(x = Age_at_Visit, y = memory, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p4 <- ggplot(mmse, aes(x = Age_at_Visit, y = orient, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p5 <- ggplot(mmse, aes(x = Age_at_Visit, y = judgment, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p6 <- ggplot(mmse, aes(x = Age_at_Visit, y = commun, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p7 <- ggplot(mmse, aes(x = Age_at_Visit, y = homehobb, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p8 <- ggplot(mmse, aes(x = Age_at_Visit, y = perscare, group = ID, colour = upvoted_cluster)) + 
  geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)
lemon::grid_arrange_shared_legend(p5, p6, p7, p8, nrow = 2, ncol = 2)

p1 <- ggplot(np, aes(x = Age_at_Visit, y = srttotal, group = ID, colour = upvoted_cluster)) + 
  geom_point() + geom_line(alpha = 0.4) + theme_bw() 
p2 <- ggplot(np, aes(x = Age_at_Visit, y = lmdelay, group = ID, colour = upvoted_cluster)) + 
  geom_point() + geom_line(alpha = 0.4) + theme_bw() + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p3 <- ggplot(np, aes(x = Age_at_Visit, y = digsym, group = ID, colour = upvoted_cluster)) + 
  geom_point() + geom_line(alpha = 0.4) + theme_bw()  + geom_smooth(method = "gam", aes(group = upvoted_cluster))
p4 <- ggplot(np, aes(x = Age_at_Visit, y = MEMUNITS, group = ID, colour = upvoted_cluster)) + 
  geom_point() + geom_line(alpha = 0.4) + theme_bw()  + geom_smooth(method = "gam", aes(group = upvoted_cluster)) +
  xlim(c(60, 85))
lemon::grid_arrange_shared_legend(p1, p2, p3, p4,  nrow = 2, ncol = 2)
```

```{r}
# mmse$memory_doubled <- mmse$commun * 2
# 
# m1 <- glmer(memory_doubled ~ 1 + 1 |ID, family = "poisson"(link = "log"), data = mmse)
# m2 <- glmer(memory_doubled ~ 1 + upvoted_cluster + 1 |ID, family = "poisson"(link = "log"), data = mmse)
# m3 <- glmer(memory_doubled ~ 1 + Age_at_Visit + 1 |ID, family = "poisson"(link = "log"), data = mmse)
# m4 <- glmer(memory_doubled ~ 1 + upvoted_cluster + Age_at_Visit + 1 |ID, family = "poisson"(link = "log"), data = mmse)
# m5 <- glmer(memory_doubled ~ 1 + upvoted_cluster * Age_at_Visit + 1 |ID, family = "poisson"(link = "log"), data = mmse)
# 
# 
# summary(glmer(memory_doubled ~ Age_at_Visit*upvoted_cluster + 1 |ID, family = "poisson"(link = "log"), data = mmse))
write.csv(df_AD_emerging_crit, ".././Data/ForNetworkPlots_AD_emerging.csv", row.names = FALSE)
write.csv(df_AD_no_crit, ".././Data/ForNetworkPlots_AD_no.csv", row.names = FALSE)
write.csv(df_emerging_no_crit, ".././Data/ForNetworkPlots_emerging_no.csv", row.names = FALSE)

```

#### Plots for drafted manuscript

```{r}
table1( ~ Age_at_LP_baseline + Age_at_LP_recent + factor(GENDER_baseline) + 
          factor(apoe_baseline) + factor(race2_baseline) + EDUC_baseline +
          LUMIPULSE_CSF_AB42_baseline + LUMIPULSE_CSF_AB42_recent + 
          LUMIPULSE_CSF_AB42_AB40_baseline + LUMIPULSE_CSF_AB42_AB40_recent + 
          pib_fsuvr_rsf_tot_cortmean_baseline + pib_fsuvr_rsf_tot_cortmean_recent + 
          Tauopathy_baseline + 
          LUMIPULSE_CSF_pTau_baseline + LUMIPULSE_CSF_pTau_recent + 
          CortSig_Thickness_baseline + CortSig_Thickness_recent + 
          WMH_volume_baseline +WMH_volume_recent +
          CSF_NfL_baseline + CSF_NfL_recent + 
          factor(cdr_baseline) + factor(cdr_recent) + 
          MMSE_baseline + MMSE_recent + 
          sumbox_baseline + sumbox_recent  | upvoted_cluster_baseline, data = df_tableone)



table1( ~ Age_at_LP_baseline + Age_at_LP_recent + factor(GENDER_baseline) + 
          factor(apoe_baseline) + factor(race2_baseline) + EDUC_baseline +
          LUMIPULSE_CSF_AB42_baseline + LUMIPULSE_CSF_AB42_recent + 
          LUMIPULSE_CSF_AB42_AB40_baseline + LUMIPULSE_CSF_AB42_AB40_recent + 
          pib_fsuvr_rsf_tot_cortmean_baseline + pib_fsuvr_rsf_tot_cortmean_recent + 
          Tauopathy_baseline + 
          LUMIPULSE_CSF_pTau_baseline + LUMIPULSE_CSF_pTau_recent + 
          CortSig_Thickness_baseline + CortSig_Thickness_recent + 
          WMH_volume_baseline +WMH_volume_recent +
          CSF_NfL_baseline + CSF_NfL_recent + 
          factor(cdr_baseline) + factor(cdr_recent) + 
          MMSE_baseline + MMSE_recent + 
          sumbox_baseline + sumbox_recent  | upvoted_cluster_baseline, 
        data = df_tableone[df_tableone$ID %in% df_pro_csf$MAPID,])

lay <- rbind(c(1, 2, 3), c(1, 2, 3), c(4, 2, 5),
             c(4, 6, 5), c(7, 6, 8), c(7, 6, 8))
g <- lemon::grid_arrange_shared_legend(pA1, pT2, pN1, pA2, pN4, pT1, pA4, pNFL1,
                           layout_matrix = lay)

ggsave(g, filename = 'Fig3a.png', dpi = 300, type = 'cairo',
       width = 8, height = 6, units = 'in')

lay <- rbind(c(1, 1, 1, 2, 2, 2), 
             c(3, 3, 3, 4, 4, 4),
             c(5, 5, 6, 6, 7, 7))
g <- lemon::grid_arrange_shared_legend(pA1,  pA4, pT2, pT1, pN1, pN4, pNFL1,
                           layout_matrix = lay)

ggsave(g, filename = 'Fig3b.png', dpi = 300, type = 'cairo',
       width = 8, height = 10, units = 'in')


df_tableone_csf <- merge(df_tableone, csf_posthoc[, c("MAPID", "TimePoint")],
                         by.x = c("ID"), by.y = c("MAPID"), all = FALSE)
CreateTableOne(vars = myVars, strata = "upvoted_cluster_recent", data = df_tableone_csf, factorVars = catVars)

#write.csv(csf, ".././Data/csf_processed_for_analysis.csv", row.names = FALSE)
```

```{r}
wmh_model <- glmer(WMH_volume ~ upvoted_cluster*Age_at_Scan + 1 |ID, family = "poisson"(link = "log"), data = wmh)
wmh$WMH_volume_log <- log(wmh$WMH_volume)
library(gamm4)
b2 <- gamm4(WMH_volume_log~s(Age_at_Scan, by = upvoted_cluster),family=gaussian,
           data=wmh[wmh$WMH_volume_log > 0,],random=~(1|ID))
plot(b2$gam,pages=1)

m1 <- bam(WMH_volume_log ~ s(Age_at_Scan, by = upvoted_cluster) + s(Age_at_Scan, by = ID), 
          data = wmh[wmh$WMH_volume_log > 0,], discrete = TRUE)
  itsadug::gamtabs(m1, type="HTML")
  
  library(mgcv)

par(mfrow=c(1,2), cex=1.1)
# version 1:
plot_smooth(m1, view="Age_at_Scan", plot_all="upvoted_cluster", rm.ranef=TRUE)

```

```{r}
library(ggpubr)
df <- merge(data.frame(csf_df), 
                      data.frame(csf_long)[, c("ID", "upvoted_cluster")], 
                      by.x = "MAPID", by.y = "ID")



my_comparisons <- list(c("AD Biomarkers", "Emerging Biomarkers"), c("AD Biomarkers", "No AD Biomarkers"),
                       c("Emerging Biomarkers, No AD Biomarkers"))

p_Amyloid_1 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Ubiquitin.fold.modifier.1", color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Ubiquitin Modifier 1 (UFM1)") +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_Amyloid_2 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Tyrosine.protein.kinase.receptor.TYRO3", color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Tyrosine Protein Kinase Receptor (TYRO3)") +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_BBB_1 <- ggboxplot(df, x = "upvoted_cluster.y", y = "SPARC.related.modular.calcium.binding.protein.1", color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("SPARC related modular calcium binding protein 1") +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.3)) + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_BBB_2 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Angiogenin" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Angiogenin" ) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.3)) + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_BBB_3 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Nidogen.2" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Nidogen - 2" ) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.3)) + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_BBB_4 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Plasminogen" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Plasminogen (plgR)" ) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.3)) + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_BBB_5 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Matrilysin" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Matrilysin" ) +
  theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.3)) + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_CVD_1 <- ggboxplot(df, x = "upvoted_cluster.y", y = "N.terminal.pro.BNP" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("N-Terminal Prohormone BNP (NT-probnp)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_inf_1 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Interleukin.17.receptor.A" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("IL-17A" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_inf_2 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Interleukin.8" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("IL-8" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_inf_3 <- ggboxplot(df, x = "upvoted_cluster.y", y = "NKG2D.ligand.3" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("NKG2D Ligand 3" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_inf_4 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Ephrin.type.B.receptor.2" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("EPH Receptor 2B" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_liver_1 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Retinol.binding.protein.4" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Retinol Binding Protein 4 (RBP 4)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_liver_2 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Hemopexin" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Hemopexin" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_liver_3 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Tumor.necrosis.factor.receptor.superfamily.member.6" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Tumor Necrosis Factor 6 (TSG6)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_1 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Calcineurin" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Calcineurin" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_2 <- ggboxplot(df, x = "upvoted_cluster.y", y = "X14.3.3.protein.family" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("14-3-3 Protein Family" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_3 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Repulsive.guidance.molecule.A" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Repulsive Guidance Molecule A" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_4 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Haptoglobin" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Haptoglobin" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_5 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Induced.myeloid.leukemia.cell.differentiation.protein.Mcl.1" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Myeloid Leukemia 1 (MCL-1)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_6 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Insulin.like.growth.factor.binding.protein.1" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Insulin Growth Factor - 1 (IGF-1)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_7 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Neurogenic.locus.notch.homolog.protein.3" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Neurogenic Locus Notch 3 (NOTCH-3)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_8 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Signal.transducer.and.activator.of.transcription.3" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("STAT-3" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_9 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Complement.C1r.subcomponent" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Complement C1r Subcomponent (C1r)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_10 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Pituitary.adenylate.cyclase.activating.polypeptide.27" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("PACAP 27" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_11 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Programmed.cell.death.1.ligand.2" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Programmed Cell Death 2 (PDL2)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_12 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Insulin.like.growth.factor.binding.protein.3" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Insulin Growth Factor - 3 (IGF-3)" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

p_ND_13 <- ggboxplot(df, x = "upvoted_cluster.y", y = "Kallikrein.8" , color = "upvoted_cluster.y",
          add = "jitter") + xlab("") + ylab("Kallikrein - 8" ) +
  theme(legend.position = "none") + scale_colour_manual(values = c("#39014A", "#10918E", "#FFE90D")) 

grid.arrange(p_Amyloid_1, p_Amyloid_2, nrow = 1)

lay <- rbind(c(1, 1, 2, 2, 3, 3), c(4, 4, 4, 5, 5, 5))
lemon::grid_arrange_shared_legend(p_BBB_1, p_BBB_2, p_BBB_3, p_BBB_4, p_BBB_5, layout_matrix = lay)

p_CVD_1

grid.arrange(p_inf_1, p_inf_2, p_inf_3, p_inf_4, nrow = 2, ncol = 2)

grid.arrange(p_liver_1, p_liver_2, p_liver_3,nrow = 1)

grid.arrange(p_ND_1, p_ND_2, p_ND_3, p_ND_4, nrow = 2, ncol = 2)
grid.arrange(p_ND_5, p_ND_6, p_ND_7, p_ND_8, nrow = 2, ncol = 2)
lemon::grid_arrange_shared_legend(p_ND_9, p_ND_10, p_ND_11, p_ND_12, p_ND_13, layout_matrix = lay)

```
